raiden/api/python.py

```diff
@@ -888,6 +893,83 @@ def channel_batch_close(
             retry_timeout=retry_timeout,
         )
 
+    def _batch_coop_settle(
+        self,
+        channels_to_settle: List[NettingChannelState],
+        retry_timeout: NetworkTimeout = DEFAULT_RETRY_TIMEOUT,
+    ) -> List[NettingChannelState]:
+        current_block = self.raiden.get_block_number()
+
+        pfs_proxy = self.raiden.pfs_proxy
+        coop_settle_state_changes: List[StateChange] = list()
+        unsuccessful_channels = list()
+        for channel_state in channels_to_settle:
+            try:
+                recipient_address = channel_state.partner_state.address
+                recipient_metadata = pfs_proxy.query_address_metadata(recipient_address)
+                if recipient_metadata:
+                    coop_settle_state_changes.append(
+                        ActionChannelCoopSettle(
+                            canonical_identifier=channel_state.canonical_identifier
+                        )
+                    )
+                else:
+                    unsuccessful_channels.append(channel_state)
+                    continue
+            except ServiceRequestFailed:
+                # node is offline / we can't get the metadata in order to communicate
+                # with the node, so don't event try an offchain coop-settle
+                unsuccessful_channels.append(channel_state)
+                continue
+        greenlets = set(self.raiden.handle_state_changes(coop_settle_state_changes))
+        gevent.joinall(greenlets, raise_error=True)
+
+        coop_settle_successful_condition = waiting.And(
+            [
+                waiting.ChannelInTargetStates([ChannelState.STATE_SETTLED]),
+                waiting.ChannelCoopSettleSuccess(self.raiden.address),
+            ]
+        )
+        channels_to_conditions = dict()
```

Comment:
Please use literal `{}`.

Reply:
> Our code style currently says otherwise but I've always disliked that.

I share the feeling. And if one looks at the bytecode difference between `s = {}` and `s = dict()` one sees
```
  2           0 BUILD_MAP                0
              2 STORE_FAST               0 (s)
```
for the first and
```
  2           0 LOAD_GLOBAL              0 (dict)
              2 CALL_FUNCTION            0
              4 STORE_FAST               0 (s)
```
for the second. In other words, using a literal translates to a specific bytecode for dict construction, whereas the other translates to a generic constructor call. Then there's the microbenchmark:
```
[ivan@beta4 ~]$ python -m timeit 's = {}'
20000000 loops, best of 5: 18.9 nsec per loop
[ivan@beta4 ~]$ python -m timeit 's = dict()'
5000000 loops, best of 5: 59.4 nsec per loop
```
which shouldn't be relevant in more than 99% cases, but still -- nice to be aware of it.